File: src/service/vehicle.service.js
import axios from "axios";

const API_URL = "http://localhost:8080";

class VehicleService{
    saveVehicles(vehicle){
        return axios.post(API_URL + "/saveVehicle", vehicle);
    }

    getAllVehicles(){
        return axios.get(API_URL + "/");
    }

    getVehicleById(id){
        return axios.get(API_URL + "/findVehicleById/" + id);
    }

    deleteVehicle(id){
        return axios.get(API_URL + "/deleteVehicle/" + id);
    }

    editVehicles(vehicle){
        return axios.post(API_URL + "/editVehicle/" + vehicle.id, vehicle);
    }

}
export default new VehicleService;

File: src/main/java/com/iuh/quy/service/VehicleService.java
package com.iuh.quy.service;

import java.util.List;

import com.iuh.quy.model.Vehicles;

public interface VehicleService {
	Vehicles saveVehicles(Vehicles vehicle);
	List<Vehicles> getAllVehicles();
	Vehicles getVehicleById(Long vehicle_id);
	String deleteVehicle(Long vehicle_id);
	Vehicles editVehicles(Vehicles vehicle, Long vehicle_id);
}
